common/src/Utilities/PString.cpp

```diff
@@ -0,0 +1,157 @@
+#include "PString.h"
+#include <codecvt>
+#include <wchar.h>
+
+PString::PString()
+{
+	count = 0;
+	string = "";
+}
+
+// Non Default
+PString::PString(const char* str)
+{
+	count = sizeof(str);
+	string = str;
+}
+
+PString::PString(const PString& rhs)
+{
+	count = rhs.count;
+	string = rhs.string;
+}
+
+PString::PString(PString&& rhs)
+{
+	count = std::move(rhs.count);
+	string = std::move(rhs.string);
+}
+
+const PString& PString::operator=(const PString& rhs)
+{
+	count = rhs.count;
+	string = rhs.string;
+	return *this;
+}
+
+PString& PString::operator=(PString& rhs)
+{
+	count = std::move(rhs.count);
+	string = std::move(rhs.string);
+}
+
+#ifdef _WIN32
+// Non default
+PString::PString(const std::wstring& str)
+{
+	count = str.size();
+	char* temp = new char[count];
+	const wchar_t* wStr = str.c_str();
+	size_t len = (wcslen(wStr) + 1) * sizeof(wchar_t);
+	wcstombs_s(&count, temp, len, wStr, len);
+	if (temp != nullptr)
```

Comment:
nullcheck too late? May try to use nullptr as input to wcstombs_s. Also, error code of wcstombs_s is not checked.